sss/sss_table.py

```diff
@@ -221,31 +340,90 @@ def check_extra_columns(df):
+            health_care = pd.concat([health_care, df['out_of_pocket']],axis=1)
+        if 'premium' in df.columns:
+            health_care = pd.concat([health_care, df['premium']],axis=1)
     else:
-        df['analysis_is_secondary'] = True
-        arpa = df[[ 'family_type','state','place','year','analysis_type']] 
-        arpa = pd.concat([arpa, df['analysis_type']],axis=1)
-        
-    return df, miscellaneous, health_care, arpa
+        df['health_care_is_secondary'] = False
+    
+    miscellaneous = pd.DataFrame()
+    if ('other_necessities'or'broadband_and_cell_phone') in df.columns:
+        df['miscellaneous_is_secondary'] = True
+        miscellaneous = df[[ 'family_type','state','place','year','analysis_type']]
+        if 'broadband_and_cell_phone' in df.columns:
+            miscellaneous = pd.concat([miscellaneous, df['broadband_and_cell_phone']],axis=1)
+        if 'other_necessities' in df.columns:
+            miscellaneous = pd.concat([miscellaneous, df['other_necessities']],axis=1)
+    else:
+        df['miscellaneous_is_secondary'] = False
+
+    return df, arpa, health_care, miscellaneous
 
 
+def prepare_for_database(df):
+    """
+    This function takes in a pandas df and prepares it for database
+
+    Specifically, this function takes in the dataframe containing 
+    data for the primary table. The dataframe is prepared by 
+    updating certain columns, adding a new weighted children column, 
+    and by dropping all duplicates. This functions is meant 
+    to update the dataframe so there are no issues transferring
+    data into the database. 
+
+
+    Parameters
+    ----------
+    df: pandas.dataframe
+        this is the dataframe containing datafor 
+
+
+    Returns
+    -------
+    pandas.datafranme
+        the returned dataframe does not contain duplicates
+
+    """
+    # handling some issues with certain columns 
+    #   (i.e., NJ 2019 has column values of "#NA")
+    df['infant'] = pd.to_numeric(df['infant'],errors='coerce')
+    df['emergency_savings'] = pd.to_numeric(
+                                df['emergency_savings'],errors='coerce')
+
+    # Create a 'weighted_child_count' column from a*c* values
+    #   this will take the infant count and move to this column
+    df['weighted_child_count'] = df['infant']
+    df.loc[df.loc[:, 'family_type'].isin(
+        [i for i in df['family_type']
+            if 'c' not in i]),
+            'weighted_child_count'] = np.nan
+
+    # removing duplicate rows
+    df = df.drop_duplicates(subset=['analysis_type', 'family_type',
+                                    'state', 'year', 'place'])
+    df.reset_index(inplace=True, drop=True)
+    return df
+
 def data_folder_to_database(data_folder, db_url=default_db_url):
     """
-    Reads path of folder of data, adds data to SQL table
+    Reads path of the data, adds data to SQL table
 
     Here, we are using a file path to loop through a folder of the SSS data.
```

Comment:
maybe make a similar generalization here.

Reply:
We might even want to change the variable name from `data_folder` to `data_path` to make it less specifically a folder? Up to you.